### PR TITLE
WIP: CMake: Better handle -fstack-protector flag support

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: cmake/EthCompilerSettings.cmake [hunk 1]
```diff
@@ -22,6 +22,13 @@ if(CCACHE_FOUND)
 	message("Using ccache")
 endif(CCACHE_FOUND)
 
+include(CheckCXXCompilerFlag)
+
+check_cxx_compiler_flag(-fstack-protector-strong stack_protector_support)
+if (stack_protector_support)
+	add_compile_options(-fstack-protector-strong)
```
Comment (axic) [inline cmake/EthCompilerSettings.cmake]: This new way doesn't pass the `-fstack-protector` option anymore as it was the case before. According to gcc it seems that `-strong` implies the basic one too. Is that correct?

Comment (chfast) [reply cmake/EthCompilerSettings.cmake]: More / less yes. The `-strong` one is stronger. The strongest is `-all`. I think this only selects what function to protect...

We are skipping an exotic case where a compiler supports `-fstack-protector` but not `-fstack-protector-strong`. I can leave with this. This does not affect solidity releases.

Comment (axic) [reply cmake/EthCompilerSettings.cmake]: Can you support both?

Comment (chfast) [reply cmake/EthCompilerSettings.cmake]: Yes. And we will. I've found that GCC 4.8 has only the basic flag.

**Comment by chriseth** [comment]
So does this PR need to be fixed or can it be merged?

**Comment by chfast** [comment]
To be fixed.

**Comment by chfast** [comment]
Fixed. Works for GCC 4.8: https://travis-ci.org/ethereum/solidity/jobs/267922031#L1261-L1264.

**Review by chriseth — APPROVED** [review]

**Pull request merged**